src/library-authoring/common/context.tsx

```diff
@@ -99,9 +104,11 @@ export type LibraryContextData = {
   // Current collection
   openCollectionInfoSidebar: (collectionId: string, additionalAction?: SidebarAdditionalActions) => void;
   // Editor modal - for editing some component
-  /** If the editor is open and the user is editing some component, this is its usageKey */
-  componentBeingEdited: string | undefined;
-  openComponentEditor: (usageKey: string) => void;
+  /** If the editor is open and the user is editing some component, this is its usageKey
+  and the callback function that should be called on closing the editor.
+  */
+  componentBeingEdited: ComponentEditorInfo | undefined;
+  openComponentEditor: (usageKey: string, onClose?: () => void) => void;
```

Comment:
nit (continued): ...and the rest of this comment can go here:

```suggestion
   // If an onClose callback is provided, it will be called when the editor is closed.
  openComponentEditor: (usageKey: string, onClose?: () => void) => void;
```